src/app/src/actions/upload.js

```diff
@@ -0,0 +1,49 @@
+import { createAction } from 'redux-act';
+
+import { logErrorAndDispatchFailure } from '../util/util';
+
+export const updateFileUploadName = createAction('UPDATE_FILE_UPLOAD_NAME');
+export const updateFileUploadDescription = createAction('UPDATE_FILE_UPLOAD_DESCRIPTION');
+export const updateFileUploadFileName = createAction('UPDATE_FILE_UPLOAD_FILE_NAME');
+
+export const startUploadFile = createAction('START_UPLOAD_FILE');
+export const failUploadFile = createAction('FAIL_UPLOAD_FILE');
+export const completeUploadFile = createAction('COMPLETE_UPLOAD_FILE');
+
+export const resetUploadState = createAction('RESET_UPLOAD_STATE');
+
+export function uploadFile(file) {
+    return (dispatch, getState) => {
+        dispatch(startUploadFile());
+
+        const {
+            upload: {
+                form: {
+                    name,
+                    description,
+                },
+            },
+        } = getState();
+
+        window.console.dir(name, description, file);
+
+        // Validate that name does not have any special char:
```

Comment:
Is this a TODO?

Reply:
I ultimately removed this altogether. The legacy implementation had the validation done client-side in the React component, but I pushed it further out and ultimately we decided that we didn't need these validations yet and could add them server-side if we ever did.